documentation-website/Writerside/topics/DAO-Entity-definition.topic

```diff
@@ -6,45 +6,228 @@
+                </def>
+                <def title="entityCtor" id="entityCtor-param">
+                    (Optional) The function that instantiates an <code>Entity</code> using the provided <code>EntityID</code>.
+                    If not provided, reflection is used to determine the primary constructor on the first access
+                    (which may affect performance).
+                </def>
+            </deflist>
+            <p>
+                In the example above, <code>IntEntityClass</code> is specified in the companion object:
+            </p>
+            <code-block lang="kotlin" src="exposed-dao/src/main/kotlin/org/example/entities/StarWarsFilmEntity.kt" include-lines="9"/>
+            <p>
+                This setup enables you to use functions provided by <code>IntEntityClass</code> to manage instances
+                of <code>StarWarsFilmEntity</code> effectively.
+            </p>
+        </chapter>
+        <chapter id="properties" title="Properties">
+            <p>
+                Each column in the table is represented as a property in the entity class, where the <code>by</code>
+                keyword ensures the data is fetched or updated from the corresponding column when accessed.
+            </p>
+            <code-block lang="kotlin" src="exposed-dao/src/main/kotlin/org/example/entities/StarWarsFilmEntity.kt"
+                        include-lines="10-13"/>
+        </chapter>
+        <p>
+            Once the entity class is defined, instances of this class allow you to manipulate individual records
+            from the corresponding table. This could involve
+            <a href="DAO-CRUD-Operations.topic" anchor="create">creating a new record</a>,
+            <a href="DAO-CRUD-Operations.topic" anchor="read">retrieving a row based on its primary key</a>,
+            <a href="DAO-CRUD-Operations.topic" anchor="update">updating values</a>, or
+            <a href="DAO-CRUD-Operations.topic" anchor="delete">deleting records</a>.
+        </p>
+    </chapter>
+    <chapter title="Immutable entities" id="immutable-entities">
+        <p>
+            For defining entities that are immutable, Exposed provides the additional
+            <a href="%BASE_API_URL%/exposed-dao/org.jetbrains.exposed.dao/-immutable-entity-class/index.html"><code>ImmutableEntityClass</code></a>
+            and
+            <a href="%BASE_API_URL%/exposed-dao/org.jetbrains.exposed.dao/-immutable-cached-entity-class/index.html"><code>ImmutableCachedEntityClass</code></a>
+            .
+        </p>
+        <p>
+            The <code>ImmutableCachedEntityClass</code> uses an internal cache to store entity loading
+            states by the associated database. This ensures that entity updates are synchronized with this class as
+            the lock object.
+        </p>
+        <p>
+            These classes can be used as companion objects when defining entities that are
+            meant to be immutable.
+        </p>
```

Comment:
Some extra info could be added here, for example, to make the point even more clear:
- You could highlight the difference between the following code block and the example in the 'Properties' section just above. Namely that properties should be defined using `val` instead of `var`.
- Because the entities are meant to be immutable, inserts should only be performed using DSL `insert()`. And updates will only be possible either via DSL `update()` or `ImmutableEntityClass.forceUpdateEntity()`. The entity class is primarily responsible for querying the associated table.

Reply:
Thanks! I added these as bullet points after the code example.